### PR TITLE
adjust to luosHAL crc reset and optism recep

### PR DESCRIPTION
**Pull request by Salem-Luos**
add crc reset

### CODE REVIEW

File: Robus/src/reception.c [hunk 1]
```diff
@@ -53,21 +53,23 @@ void Recep_Init(void)
  ******************************************************************************/
 void Recep_GetHeader(volatile unsigned char *data)
 {
-    ctx.tx_lock = true;
```
Comment (nicolas-rabault) [inline Robus/src/reception.c:56]: This is not related to CRC but to TX_lock

File: Robus/src/reception.c [hunk 2]
```diff
@@ -77,34 +79,36 @@ void Recep_GetHeader(volatile unsigned char *data)
...
+                {
+                    data_size = current_msg->header.size;
+                }
+                if (keep)
+                {
+                    if (data_size)
+                    {
+                        MsgAlloc_ValidHeader();
+                    }
+                }
```
Comment (nicolas-rabault) [inline Robus/src/reception.c:101]:
```suggestion
                if (keep && data_size)
                {
                    MsgAlloc_ValidHeader();
                }
```

File: Robus/src/reception.c [hunk 3]
```diff
@@ -77,34 +79,36 @@ void Recep_GetHeader(volatile unsigned char *data)
...
+                else
+                {
+                    MsgAlloc_InvalidMsg();
+                }
+            break;
+
+        default:
+        break;
     }
+    LuosHAL_ComputeCRC((uint8_t *)data, 1, (uint8_t *)&crc_val);
```
Comment (nicolas-rabault) [inline Robus/src/reception.c:111]: Because we have to check and reload context at each TX we have to compute only byte by byte. We should remove the size.

**Review by nicolas-rabault — CHANGES_REQUESTED** [review]
This PR can't be merged for now. We could have conflict between TX and RX CRC computation. The TX side have to take care of it by verifying reception state and reloading the CRC context.

File: Robus/src/reception.c (hunk 1 above)
Comment (Salem-Luos) [reply Robus/src/reception.c:56]: still there just change position

File: Robus/src/reception.c (hunk 2 above)
Comment (Salem-Luos) [reply Robus/src/reception.c:101]: not possible

**Review by nicolas-rabault — APPROVED** [review]
There is a typo error in : "adjust to luosHAL crc reset and optism recep"
But you have to fixup both commit here.
It good for me.

**Pull request merged**